modules/google-maps/src/utils.js

```diff
@@ -0,0 +1,155 @@
+/* global document, google */
+import {Deck} from '@deck.gl/core';
+
+/**
+ * Get a new deck instance
+ * @param map (google.maps.Map) - The parent Map instance
+ * @param overlay (google.maps.OverlayView) - A maps Overlay instance
+ * @param [deck] (Deck) - a previously created instances
+ */
+export function createDeckInstance(map, overlay, deck) {
+  if (deck) {
+    if (deck.props.userData._googleMap === map) {
+      return deck;
+    }
+    // deck instance was created for a different map
+    destroyDeckInstance(deck);
+  }
+
+  const eventListeners = {
+    click: null,
+    mousemove: null,
+    mouseout: null
+  };
+
+  deck = new Deck({
+    canvas: createDeckCanvas(overlay),
+    initialViewState: {
+      longitude: 0,
+      latitude: 0,
+      zoom: 1
+    },
+    controller: false,
+    userData: {
+      _googleMap: map,
+      _eventListeners: eventListeners
+    }
+  });
+
+  // Register event listeners
+  for (const eventType in eventListeners) {
+    eventListeners[eventType] = map.addListener(eventType, evt =>
+      handleMouseEvent(deck, eventType, evt)
+    );
+  }
+
+  return deck;
+}
+
+function createDeckCanvas(overlay) {
+  // google maps places overlays in a container anchored at the map center.
+  // the container CSS is manipulated during dragging,
+  // making it difficult for our canvas to sync with the map.
+  // As a hack, we find that container and append our canvas as its sibling.
+  let container = overlay.getPanes().overlayLayer;
+  while (container && container.style.left !== '50%') {
+    container = container.offsetParent;
+  }
+  container = container.offsetParent;
```

Comment:
This gist is a self-contained example where the canvas can be appended as a child of the `overlayLayer`, rather than traversing its ancestors: 

https://gist.github.com/donmccurdy/61f94f91b60682bf325d2cceae4c70f2

That feels less fragile to me – I know you were seeing some tearing when you implemented it that way, but in the case in this gist I can't reproduce that.

Reply:
I see what you are doing. Confirmed it works. I was using `map.getBounds()` to determine the northwestern corner which was incorrect.